components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,136 @@
+                orgName={org?.slug ?? org?.id}
+                attributionId={attributionId}
+                startDate={startDate}
+                endDate={endDate}
+            />,
+            {
+                autoHide: false,
+            },
+        );
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" onClick={handleDownload} className="flex flex-row" icon={<DownloadIcon />}>
+            Export as CSV
+        </Button>
+    );
+};
+
+type DownloadUsageToastProps = Props & {
+    orgName: string;
+};
+
+const DownloadUsageToast: FC<DownloadUsageToastProps> = ({ attributionId, endDate, startDate, orgName }) => {
+    const [progress, setProgress] = useState(0);
+
+    const queryArgs = useMemo(
+        () => ({
+            orgName,
+            attributionId: AttributionId.render(attributionId),
+            from: startDate.startOf("day").valueOf(),
+            to: endDate.endOf("day").valueOf(),
+            onProgress: setProgress,
+        }),
+        [attributionId, endDate, orgName, startDate],
+    );
+    const { data, error, isLoading, abort, remove } = useDownloadUsageCSV(queryArgs);
+
+    const saveFile = useCallback(() => {
+        if (!data || !data.blob) {
+            return;
+        }
+
+        saveAs(data.blob, data.filename);
+    }, [data]);
+
+    useEffect(() => {
+        return () => {
+            abort();
+            remove();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    if (isLoading) {
+        return (
+            <div className="space-y-2">
+                <div className="flex flex-row items-center space-x-2">
+                    <span>Preparing usage export</span>
+                </div>
+                <ProgressBar percent={progress} />
+            </div>
+        );
+    }
+
+    if (error) {
+        return (
+            <div className="flex flex-row items-start space-x-2">
+                <ExclamationIcon className="w-5 h-5 mt-0.5" />
+
+                <div>
+                    <span>Error exporting your usage data:</span>
+                    <pre className="mt-2 whitespace-normal text-sm">{error.message}</pre>
+                </div>
+            </div>
+        );
+    }
+
+    if (!data || !data.blob || data.count === 0) {
+        return <span>No usage data for the selected period.</span>;
+    }
+
+    const readableSize = prettyBytes(data.blob.size);
+    const formattedCount = Intl.NumberFormat().format(data.count);
+
+    return (
+        <div className="flex flex-row items-start justify-between space-x-2">
+            <div>
+                <span>Usage export complete.</span>
+                <p>
+                    {readableSize} &middot; {formattedCount} {data.count !== 1 ? "entries" : "entry"} exported
+                </p>
+            </div>
+            <div>
+                <LinkButton onClick={saveFile} className="text-left text-base">
```

Comment:
**issue(non-blocking)**: There's a minor accessibility issue with the link color as the logic is reverted—not sure if it's worth the effort, but the accessibility issue is there, affecting both light and dark themes.

| CURRENT | CORRECT |
|-|-|
| <img width="464" alt="Frame 1568" src="https://github.com/gitpod-io/gitpod/assets/120486/ffe8a3c4-3e93-4a81-ad93-1ed2dac80a46"> |  <img width="464" alt="Frame 1567" src="https://github.com/gitpod-io/gitpod/assets/120486/7a08606e-5da7-4518-a6d9-de30a93d9372"> |

Reply:
Should it be that color for both light and dark? I think I used whatever color our `gp-link` class was using, but I’ll double check and change to make it that lighter shade.

Reply:
@selfcontained Ideally, it would be `blue-400` on dark backgrounds and `blue-500`. The `gp-link` is using these values, but the problem here is that the dark-colored toast is shown on the light them, and it's using `blue-400` instead of `blue-500` and vice versa for the dark theme. Hopefully this makes sense. 😅

Reply:
oh right! Now I follow. Yah that toasts being almost an inverted theme does make it interesting to figure out the right way to handle things!